core/src/main/scala/com/microsoft/azure/synapse/ml/fabric/FabricClient.scala

```diff
@@ -0,0 +1,154 @@
+  var MLWorkloadHost = "";
+
+  private val WorkloadEndpointTypeML = "ML";
+  private val WorkloadEndpointTypeLLMPlugin = "LlmPlugin"
+  private val WorkloadEndpointTypeAutomatic = "Automatic"
+  private val WorkloadEndpointTypeRegistry = "Registry"
+  private val WorkloadEndpointTypeAdmin = "MLAdmin"
+
+  lazy val PbiSharedHost: String = getPbiSharedHost;
+
+  lazy val MLWorkloadEndpointML: String = getMLWorkloadEndpoint(WorkloadEndpointTypeML);
+  lazy val MLWorkloadEndpointLLMPlugin: String = getMLWorkloadEndpoint(WorkloadEndpointTypeLLMPlugin);
+  lazy val MLWorkloadEndpointAutomatic: String = getMLWorkloadEndpoint(WorkloadEndpointTypeAutomatic);
+  lazy val MLWorkloadEndpointRegistry: String = getMLWorkloadEndpoint(WorkloadEndpointTypeRegistry);
+  lazy val MLWorkloadEndpointAdmin: String = getMLWorkloadEndpoint(WorkloadEndpointTypeAdmin);
+  lazy val MLWorkloadEndpointCognitive: String = s"${MLWorkloadEndpointML}cognitive/";
+  lazy val MLWorkloadEndpointOpenAI: String = s"${MLWorkloadEndpointML}cognitive/openai/";
+
+  private val ContextFilePath = "/home/trusted-service-user/.trident-context";
+  private val SparkConfPath = "/opt/spark/conf/spark-defaults.conf";
+
+  lazy val MyLogger: Logger = Logger.getLogger(this.getClass.getName);
+
+  if(PlatformDetails.runningOnFabric()) {
+    readFabricContextFile();
+    readFabricSparkConfFile();
+    init();
+  }
+
+  def init(): Unit ={
+    this.CapacityID = this.FabricContext.getOrElse("trident.capacity.id", "");
+    this.WorkspaceID = this.FabricContext.getOrElse("trident.artifact.workspace.id", "");
+    this.ArtifactID = this.FabricContext.getOrElse("trident.artifact.id", "");
+    this.PbiEnv = this.FabricContext.getOrElse("spark.trident.pbienv", "public").toLowerCase();
+    this.MLWorkloadHost = this.extractSchemeAndHost(
+      this.FabricContext.getOrElse("trident.lakehouse.tokenservice.endpoint", "https://")
+    ).getOrElse("");
+  }
+
+  def extractSchemeAndHost(urlString: String): Option[String] = {
+    try {
+      val url = new URL(urlString)
+      val scheme = url.getProtocol
+        val host = url.getHost
+        Some(s"$scheme://$host")
+    } catch {
+      case _: Exception =>
+        // Handle MalformedURLException or other exceptions
+        None
+    }
+  }
+
+  def readFabricContextFile(): Unit = {
+    Try(Source.fromFile(ContextFilePath)) match {
+      case Success(source) =>
+        try {
+          for (line <- source.getLines()) {
+            val keyValue = line.split('=')
+            if (keyValue.length == 2) {
+              val Array(k, v) = keyValue.map(_.trim)
+              FabricContext += (k -> v)
+            }
+          }
+          source.close()
+        } catch {
+          case e: Exception =>
+            MyLogger.error("Error reading Fabric context file", e)
+        }
+      case Failure(exception) =>
+        MyLogger.error("Error opening Fabric context file", exception)
+    }
+  }
```

Comment:
to avoid mutable datastructures (which are discouraged in scala (also i know this is a python rewrite and im sorry lol)) you can do something like 
```suggestion
  def readFabricContextFile(): Unit = {
    Try(Source.fromFile(ContextFilePath)) match {
      case Success(source) =>
        val enties = source.getLines().map(_.split("=")).filter(_.length == 2).map(entries =>
            Tuple2(entries.map(entry => entry.trim))
        ).toMap
        source.close()
        entries
      case Failure(exception) =>
        MyLogger.error("Error opening Fabric context file", exception)
    }
  }
```

Reply:
Also i think you dont need to make a catchall error statement here